drivers/charger/charger_pf1550.c

```diff
@@ -0,0 +1,695 @@
+		return ret;
+	}
+
+	ret = pf1550_led_config(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to configure led: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int pf1550_get_prop(const struct device *dev, charger_prop_t prop,
+			     union charger_propval *val)
+{
+	struct charger_pf1550_data *data = dev->data;
+
+	switch (prop) {
+	case CHARGER_PROP_ONLINE:
+		val->online = data->charger_online;
+		return 0;
+	case CHARGER_PROP_STATUS:
+		val->status = data->charger_status;
+		return 0;
+	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
+		val->const_charge_current_ua = data->charge_current_ua;
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int pf1550_set_prop(const struct device *dev, charger_prop_t prop,
+			     const union charger_propval *val)
+{
+	struct charger_pf1550_data *data = dev->data;
+	int ret;
+
+	switch (prop) {
+	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
+		ret = pf1550_set_constant_charge_current(dev, val->const_charge_current_ua);
+		if (ret == 0) {
+			data->charge_current_ua = val->const_charge_current_ua;
+		}
+		return ret;
+	case CHARGER_PROP_INPUT_REGULATION_CURRENT_UA:
+		ret = pf1550_set_vbus_ilim(dev, val->input_current_regulation_current_ua);
+		if (ret == 0) {
+			data->vbus_ilim_ua = val->input_current_regulation_current_ua;
+		}
+		return ret;
+	case CHARGER_PROP_STATUS_NOTIFICATION:
+		data->charger_status_notifier = val->status_notification;
+		return 0;
+	case CHARGER_PROP_ONLINE_NOTIFICATION:
+		data->charger_online_notifier = val->online_notification;
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int pf1550_enable_interrupt_pin(const struct device *dev, bool enabled)
+{
+	const struct charger_pf1550_config *const config = dev->config;
+	gpio_flags_t flags;
+	int ret;
+
+	flags = enabled ? GPIO_INT_EDGE_TO_ACTIVE : GPIO_INT_DISABLE;
+
+	ret = gpio_pin_interrupt_configure_dt(&config->int_gpio, flags);
+	if (ret < 0) {
+		LOG_ERR("Could not %s interrupt GPIO callback: %d", enabled ? "enable" : "disable",
+			ret);
+	}
+
+	return ret;
+}
+
+static void pf1550_gpio_callback(const struct device *dev, struct gpio_callback *cb,
+				   uint32_t pins)
+{
+	struct charger_pf1550_data *data = CONTAINER_OF(cb, struct charger_pf1550_data,
+							  gpio_cb);
+	int ret;
+
+	(void) pf1550_enable_interrupt_pin(data->dev, false);
+
+	ret = k_work_submit(&data->int_routine_work);
+	if (ret < 0) {
+		LOG_WRN("Could not submit int work: %d", ret);
+	}
+}
+
+static void pf1550_int_routine_work_handler(struct k_work *work)
+{
+	struct charger_pf1550_data *data = CONTAINER_OF(work, struct charger_pf1550_data,
+							  int_routine_work);
+	uint8_t int_src;
+	int ret;
+
+	ret = pf1550_get_interrupt_source(data->dev, &int_src);
+	if (ret < 0) {
+		LOG_WRN("Failed to read interrupt source");
+		return;
+	}
+
+	LOG_DBG("Interrupt received: %x",int_src);
+
+	ret = pf1550_get_charger_status(data->dev, &data->charger_status);
+	if (ret < 0) {
+		LOG_WRN("Failed to read charger status: %d", ret);
```

Comment:
do you really want to try and go ahead with the other calls here? if `pf1550_get_charger_status` likely the whole i2c bus is hosed and there's no hope for any of the other calls to succseed ever, may be better off just `return ret` and keep the rest of the code inline